Web Inspector: drop InspectorAdditionsEnabled in favor of new methods on InspectorFrontendhost

dcrousso opened the pull request:
#### bc3d1ea18a109095c934af5c547b9e9bd2c7ee1c
<pre>
Web Inspector: drop InspectorAdditionsEnabled in favor of new methods on InspectorFrontendhost
<a href="https://bugs.webkit.org/show_bug.cgi?id=241192">https://bugs.webkit.org/show_bug.cgi?id=241192</a>

Reviewed by Patrick Angle.

This removes custom things from standardized IDLs and allows custom functionality necessary for Web
Inspector to be kept inside inspector objects (as well as the fact that `InspectorFrontendHost` is
guaranteed to exist, so there&apos;s no concern about this functionality not being available).

* Source/WTF/Scripts/Preferences/WebPreferences.yaml:
* Source/WebCore/page/RuntimeEnabledFeatures.h:
(WebCore::RuntimeEnabledFeatures::setInspectorAdditionsEnabled): Deleted.
(WebCore::RuntimeEnabledFeatures::inspectorAdditionsEnabled const): Deleted.
* Source/WebKit/UIProcess/API/C/WKPreferences.cpp:
(WKPreferencesSetInspectorAdditionsEnabled): Deleted.
(WKPreferencesGetInspectorAdditionsEnabled): Deleted.
* Source/WebKit/UIProcess/API/C/WKPreferencesRefPrivate.h:
* Source/WebKit/WebProcess/Inspector/WebInspectorUI.cpp:
(WebKit::WebInspectorUI::enableFrontendFeatures):
* Source/WebKitLegacy/mac/WebView/WebPreferenceKeysPrivate.h:
* Source/WebKitLegacy/mac/WebView/WebPreferences.mm:
(-[WebPreferences inspectorAdditionsEnabled]): Deleted.
(-[WebPreferences setInspectorAdditionsEnabled:]): Deleted.
* Source/WebKitLegacy/mac/WebView/WebPreferencesPrivate.h:
* Source/WebKitLegacy/win/Interfaces/IWebPreferencesPrivate.idl:
* Source/WebKitLegacy/win/WebPreferenceKeysPrivate.h:
* Source/WebKitLegacy/win/WebPreferences.cpp:
(WebPreferences::initializeDefaultSettings):
(WebPreferences::inspectorAdditionsEnabled): Deleted.
(WebPreferences::setInspectorAdditionsEnabled): Deleted.
* Source/WebKitLegacy/win/WebPreferences.h:
* Source/WebKitLegacy/win/WebView.cpp:
(WebView::notifyPreferencesChanged):
* Tools/DumpRenderTree/TestOptions.cpp:
(WTR::TestOptions::defaults):
Remove all things related to `InspectorAdditionsEnabled`.

* Source/WebCore/html/canvas/CanvasPath.idl:
* Source/WebCore/html/canvas/CanvasRenderingContext2D.idl:
* Source/WebCore/html/canvas/OffscreenCanvasRenderingContext2D.idl:
Remove nonstandard functionality gated by `InspectorAdditionsEnabled` in standardized IDLs.

* Source/WebCore/inspector/InspectorFrontendHost.h:
* Source/WebCore/inspector/InspectorFrontendHost.idl:
* Source/WebCore/inspector/InspectorFrontendHost.cpp:
(WebCore::InspectorFrontendHost::getPath const): Added.
(WebCore::InspectorFrontendHost::getCurrentX const): Added.
(WebCore::InspectorFrontendHost::getCurrentY const): Added.
(WebCore::InspectorFrontendHost::setPath const): Added.
* Source/WebInspectorUI/UserInterface/Base/IDLExtensions.js: Added.
(File.prototype.getPath): Added.
(CanvasRenderingContext2D.prototype.currentX): Added.
(CanvasRenderingContext2D.prototype.currentY): Added.
(CanvasRenderingContext2D.prototype.getPath): Added.
(CanvasRenderingContext2D.prototype.setPath): Added.
Centralize all custom IDL extensions in a new file with the same name so it&apos;s very easy to find.

* Source/WebInspectorUI/UserInterface/Base/ImageUtilities.js:
(WI.ImageUtilities.supportsCanvasPathDebugging): Deleted.
* Source/WebInspectorUI/UserInterface/Models/RecordingAction.js:
(WI.RecordingAction.prototype.process):
* Source/WebInspectorUI/UserInterface/Models/RecordingState.js:
(WI.RecordingState.prototype.fromContext):
* Source/WebInspectorUI/UserInterface/Views/RecordingContentView.js:
(WI.RecordingContentView):
(WI.RecordingContentView.prototype.get navigationItems):
(WI.RecordingContentView.prototype.attached):
(WI.RecordingContentView.prototype._generateContentCanvas2D):
* Source/WebInspectorUI/UserInterface/Views/ResourceContentView.js:
(WI.ResourceContentView.prototype.async _handleMapLocalResourceOverrideToFile):
Remove checks based on `InspectorAdditionsEnabled` (technically they check for the existence of the
custom functionality on the builtin prototype, but that&apos;s added based on `InspectorAdditionsEnabled`).

* Source/WebInspectorUI/UserInterface/Main.html:
* Source/WebInspectorUI/UserInterface/Test.html:

* LayoutTests/fast/canvas/2d.currentPoint.html: Removed.
* LayoutTests/fast/canvas/2d.currentPoint-expected.txt: Removed.
* LayoutTests/fast/canvas/2d.getPath.modification.html: Removed.
* LayoutTests/fast/canvas/2d.getPath.modification-expected.txt: Removed.
* LayoutTests/fast/canvas/2d.getPath.newobject.html: Removed.
* LayoutTests/fast/canvas/2d.getPath.newobject-expected.txt: Removed.
* LayoutTests/fast/canvas/2d.setPath.html: Removed.
* LayoutTests/fast/canvas/2d.setPath-expected.txt: Removed.
* LayoutTests/inspector/idl-extensions/CanvasRenderingContext2D/getPath.html: Added.
* LayoutTests/inspector/idl-extensions/CanvasRenderingContext2D/getPath-expected.txt: Added.
* LayoutTests/inspector/idl-extensions/CanvasRenderingContext2D/get_currentX.html: Added.
* LayoutTests/inspector/idl-extensions/CanvasRenderingContext2D/get_currentX-expected.txt: Added.
* LayoutTests/inspector/idl-extensions/CanvasRenderingContext2D/get_currentY.html: Added.
* LayoutTests/inspector/idl-extensions/CanvasRenderingContext2D/get_currentY-expected.txt: Added.
* LayoutTests/inspector/idl-extensions/CanvasRenderingContext2D/setPath.html: Added.
* LayoutTests/inspector/idl-extensions/CanvasRenderingContext2D/setPath-expected.txt: Added.

Canonical link: <a href="https://commits.webkit.org/251287@main">https://commits.webkit.org/251287@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@295232">https://svn.webkit.org/repository/webkit/trunk@295232</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

JosephPecoraro commented on Source/WebInspectorUI/UserInterface/Base/IDLExtensions.js:
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2017 Apple Inc. All rights reserved.
```
`2022`?

JosephPecoraro commented on Source/WebInspectorUI/UserInterface/Base/IDLExtensions.js:
```diff
@@ -0,0 +1,58 @@
+
+Object.defineProperty(CanvasRenderingContext2D.prototype, "setPath", {
+    value(path) {
+        InspectorFrontendHost.setPath(this, path);
```
Probably should still `return` the result, even though nothing is expected?

JosephPecoraro commented on Source/WebCore/inspector/InspectorFrontendHost.cpp:
```diff
@@ -781,5 +778,29 @@ ExceptionOr<JSC::JSValue> InspectorFrontendHost::evaluateScriptInExtensionTab(HT
+    return context.getPath();
+}
+
+void InspectorFrontendHost::setPath(CanvasRenderingContext2D& context, const Path2D& path) const
```
Drop the `const`? Weird to have a "const" setter.

patrickangle commented on Source/WebCore/inspector/InspectorFrontendHost.idl:
```diff
@@ -106,6 +104,15 @@
+    float getCurrentX(CanvasRenderingContext2D canvasPath);
+    float getCurrentY(CanvasRenderingContext2D canvasPath);
+    [NewObject] Path2D getPath(CanvasRenderingContext2D context);
+    undefined setPath(CanvasRenderingContext2D context, Path2D path);
```
Should these functions have something in their name to indicate they operate on canvas contexts? It seems odd to me that we have two different `getPath` functions with very different purposes. Seems like a recipe for confusion. (Full review coming later, but noticed this and wanted to ask.)

dcrousso replied:
maybe?  IDL does allow for overloading, so it's totally allowed.

The more important issue tho is that the backend (specifically `InspectorCanvas::buildInitialState`) uses `"setPath"` (as a string) to encode this information for use by the frontend, meaning that if we wanted to change the name we'd have to adjust the backend (and add backwards compatibility), which I'd prefer to avoid if possible.

patrickangle commented on Source/WebCore/inspector/InspectorFrontendHost.idl:
```diff
@@ -109,6 +107,15 @@
+    DOMString getPath(File file);
+
+    float getCurrentX(CanvasRenderingContext2D canvasPath);
+    float getCurrentY(CanvasRenderingContext2D canvasPath);
```
Ditto InspectorFrontendHost.cpp:823

patrickangle commented on Source/WebCore/inspector/InspectorFrontendHost.cpp:
```diff
@@ -818,5 +815,29 @@ ExceptionOr<JSC::JSValue> InspectorFrontendHost::evaluateScriptInExtensionTab(HT
+    return file.path();
+}
+
+float InspectorFrontendHost::getCurrentX(const CanvasRenderingContext2D& path) const
```
`const CanvasRenderingContext2D& context`?

patrickangle commented on Source/WebCore/inspector/InspectorFrontendHost.cpp:
```diff
@@ -818,5 +815,29 @@ ExceptionOr<JSC::JSValue> InspectorFrontendHost::evaluateScriptInExtensionTab(HT
+    return path.currentX();
+}
+
+float InspectorFrontendHost::getCurrentY(const CanvasRenderingContext2D& path) const
```
Ditto :823

patrickangle commented on Source/WebInspectorUI/UserInterface/Base/ImageUtilities.js:
```diff
@@ -126,14 +126,6 @@ WI.ImageUtilities = class ImageUtilities {
         return image;
     }
-
-    static supportsCanvasPathDebugging()
-    {
-        return "getPath" in CanvasRenderingContext2D.prototype
-            && "setPath" in CanvasRenderingContext2D.prototype
-            && "currentX" in CanvasRenderingContext2D.prototype
-            && "currentY" in CanvasRenderingContext2D.prototype;
-    }
```
Aside: under what circumstance would this have not returned false before? Should we ever have had an instance of Web Inspector where the Web Inspector Additions weren't enabled?

dcrousso replied:
i think WebKitLegacy didn't have this enabled

ryanhaddad committed:
Repurpose arm64 iOS simulator GPUP bot as a general arm64 iOS simulator tester
https://bugs.webkit.org/show_bug.cgi?id=241278

Reviewed by Aakash Jain.

* Tools/CISupport/build-webkit-org/config.json:
* Tools/CISupport/build-webkit-org/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/build-webkit-org/public_html/dashboard/Scripts/WebKitBuildbot.js:
(WebKitBuildbot):

Canonical link: https://commits.webkit.org/251284@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295229 268f45cc-cd09-0410-ab3c-d52691b4dbfc

achristensen07 committed:
HSTS synthesized redirect responses should not be blocked by CORS
https://bugs.webkit.org/show_bug.cgi?id=241003

Reviewed by Youenn Fablet.

If a cross-origin request is made to an http URL that would be upgraded to an https URL
because of HSTS, we synthesize a "response" to call willPerformHTTPRedirection with.
Unfortunately, this response can fail CORS checks causing the request to be unnecessarily blocked.
To prevent this, just add CORS headers to the synthesized responses for HSTS.

* Source/WebKit/NetworkProcess/cocoa/NetworkSessionCocoa.mm:
(-[WKNetworkSessionDelegate URLSession:task:_schemeUpgraded:completionHandler:]):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/HSTS.mm:
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/251285@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295230 268f45cc-cd09-0410-ab3c-d52691b4dbfc

hortont424 committed:
Unreviewed build fix after r295149

* Source/WebKit/Shared/RemoteLayerTree/CGDisplayListImageBufferBackend.cpp:
(WebKit::CGDisplayListImageBufferBackend::getPixelBuffer const):
* Source/WebKit/Shared/RemoteLayerTree/CGDisplayListImageBufferBackend.h:
Fix the build.

Canonical link: https://commits.webkit.org/251286@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295231 268f45cc-cd09-0410-ab3c-d52691b4dbfc

dcrousso committed:
Web Inspector: drop InspectorAdditionsEnabled in favor of new methods on InspectorFrontendhost
https://bugs.webkit.org/show_bug.cgi?id=241192

Reviewed by Patrick Angle.

This removes custom things from standardized IDLs and allows custom functionality necessary for Web
Inspector to be kept inside inspector objects (as well as the fact that `InspectorFrontendHost` is
guaranteed to exist, so there's no concern about this functionality not being available).

* Source/WTF/Scripts/Preferences/WebPreferences.yaml:
* Source/WebCore/page/RuntimeEnabledFeatures.h:
(WebCore::RuntimeEnabledFeatures::setInspectorAdditionsEnabled): Deleted.
(WebCore::RuntimeEnabledFeatures::inspectorAdditionsEnabled const): Deleted.
* Source/WebKit/UIProcess/API/C/WKPreferences.cpp:
(WKPreferencesSetInspectorAdditionsEnabled): Deleted.
(WKPreferencesGetInspectorAdditionsEnabled): Deleted.
* Source/WebKit/UIProcess/API/C/WKPreferencesRefPrivate.h:
* Source/WebKit/WebProcess/Inspector/WebInspectorUI.cpp:
(WebKit::WebInspectorUI::enableFrontendFeatures):
* Source/WebKitLegacy/mac/WebView/WebPreferenceKeysPrivate.h:
* Source/WebKitLegacy/mac/WebView/WebPreferences.mm:
(-[WebPreferences inspectorAdditionsEnabled]): Deleted.
(-[WebPreferences setInspectorAdditionsEnabled:]): Deleted.
* Source/WebKitLegacy/mac/WebView/WebPreferencesPrivate.h:
* Source/WebKitLegacy/win/Interfaces/IWebPreferencesPrivate.idl:
* Source/WebKitLegacy/win/WebPreferenceKeysPrivate.h:
* Source/WebKitLegacy/win/WebPreferences.cpp:
(WebPreferences::initializeDefaultSettings):
(WebPreferences::inspectorAdditionsEnabled): Deleted.
(WebPreferences::setInspectorAdditionsEnabled): Deleted.
* Source/WebKitLegacy/win/WebPreferences.h:
* Source/WebKitLegacy/win/WebView.cpp:
(WebView::notifyPreferencesChanged):
* Tools/DumpRenderTree/TestOptions.cpp:
(WTR::TestOptions::defaults):
Remove all things related to `InspectorAdditionsEnabled`.

* Source/WebCore/html/canvas/CanvasPath.idl:
* Source/WebCore/html/canvas/CanvasRenderingContext2D.idl:
* Source/WebCore/html/canvas/OffscreenCanvasRenderingContext2D.idl:
Remove nonstandard functionality gated by `InspectorAdditionsEnabled` in standardized IDLs.

* Source/WebCore/inspector/InspectorFrontendHost.h:
* Source/WebCore/inspector/InspectorFrontendHost.idl:
* Source/WebCore/inspector/InspectorFrontendHost.cpp:
(WebCore::InspectorFrontendHost::getPath const): Added.
(WebCore::InspectorFrontendHost::getCurrentX const): Added.
(WebCore::InspectorFrontendHost::getCurrentY const): Added.
(WebCore::InspectorFrontendHost::setPath const): Added.
* Source/WebInspectorUI/UserInterface/Base/IDLExtensions.js: Added.
(File.prototype.getPath): Added.
(CanvasRenderingContext2D.prototype.currentX): Added.
(CanvasRenderingContext2D.prototype.currentY): Added.
(CanvasRenderingContext2D.prototype.getPath): Added.
(CanvasRenderingContext2D.prototype.setPath): Added.
Centralize all custom IDL extensions in a new file with the same name so it's very easy to find.

* Source/WebInspectorUI/UserInterface/Base/ImageUtilities.js:
(WI.ImageUtilities.supportsCanvasPathDebugging): Deleted.
* Source/WebInspectorUI/UserInterface/Models/RecordingAction.js:
(WI.RecordingAction.prototype.process):
* Source/WebInspectorUI/UserInterface/Models/RecordingState.js:
(WI.RecordingState.prototype.fromContext):
* Source/WebInspectorUI/UserInterface/Views/RecordingContentView.js:
(WI.RecordingContentView):
(WI.RecordingContentView.prototype.get navigationItems):
(WI.RecordingContentView.prototype.attached):
(WI.RecordingContentView.prototype._generateContentCanvas2D):
* Source/WebInspectorUI/UserInterface/Views/ResourceContentView.js:
(WI.ResourceContentView.prototype.async _handleMapLocalResourceOverrideToFile):
Remove checks based on `InspectorAdditionsEnabled` (technically they check for the existence of the
custom functionality on the builtin prototype, but that's added based on `InspectorAdditionsEnabled`).

* Source/WebInspectorUI/UserInterface/Main.html:
* Source/WebInspectorUI/UserInterface/Test.html:

* LayoutTests/fast/canvas/2d.currentPoint.html: Removed.
* LayoutTests/fast/canvas/2d.currentPoint-expected.txt: Removed.
* LayoutTests/fast/canvas/2d.getPath.modification.html: Removed.
* LayoutTests/fast/canvas/2d.getPath.modification-expected.txt: Removed.
* LayoutTests/fast/canvas/2d.getPath.newobject.html: Removed.
* LayoutTests/fast/canvas/2d.getPath.newobject-expected.txt: Removed.
* LayoutTests/fast/canvas/2d.setPath.html: Removed.
* LayoutTests/fast/canvas/2d.setPath-expected.txt: Removed.
* LayoutTests/inspector/idl-extensions/CanvasRenderingContext2D/getPath.html: Added.
* LayoutTests/inspector/idl-extensions/CanvasRenderingContext2D/getPath-expected.txt: Added.
* LayoutTests/inspector/idl-extensions/CanvasRenderingContext2D/get_currentX.html: Added.
* LayoutTests/inspector/idl-extensions/CanvasRenderingContext2D/get_currentX-expected.txt: Added.
* LayoutTests/inspector/idl-extensions/CanvasRenderingContext2D/get_currentY.html: Added.
* LayoutTests/inspector/idl-extensions/CanvasRenderingContext2D/get_currentY-expected.txt: Added.
* LayoutTests/inspector/idl-extensions/CanvasRenderingContext2D/setPath.html: Added.
* LayoutTests/inspector/idl-extensions/CanvasRenderingContext2D/setPath-expected.txt: Added.

Canonical link: https://commits.webkit.org/251287@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295232 268f45cc-cd09-0410-ab3c-d52691b4dbfc
